### PR TITLE
Add stateful application tutorial.

### PR DESCRIPTION
**Pull request by jeffmendoza**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1599)

### CODE REVIEW

**Comment by jeffmendoza** [comment]
Preview looks good.
@steveperry-53 @devin-donnelly please review. Thanks!

**Comment by jeffmendoza** [comment]
@askcarter @waprin PTAL

**Comment by waprin** [comment]
How do we stage it again?

**Comment by jeffmendoza** [comment]
@waprin you can check out this pr, under the "All checks", then click Details next to deploy/netlify

To stage your stuff before opening a PR, check out https://github.com/kubernetes/kubernetes.github.io#staging-the-site-locally-using-docker

**Comment by steveperry-53** [comment]
I'll take a look at this and provide feedback by EOD Friday 11/4.

**Comment by steveperry-53** [comment]
I really like the overall tutorial and its flow. I learned new things easily by going through this tutorial. I added that has some suggested changes and also brings the document into alignment with our style guide.

**Comment by steveperry-53** [comment]
Investigating why the deploy/netlify check failed.

File: docs/tutorials/stateful-application/run-stateful-application.md [hunk 1]
```diff
@@ -21,50 +21,36 @@ application we will use is MySQL.
...
-  configuration is done automatically if you use Minikube.
-
-* For data persistence we will create a Persistent Volume that
-  references a disk in your
-  environment. See
-  [here](/docs/user-guide/persistent-volumes/#types-of-persistent-volumes) for
-  the types of environments supported. This Tutorial will demonstrate
-  `GCEPersistentDisk` but any type will work. `GCEPersistentDisk`
-  volumes only work on Google Compute Engine.
+* {% include task-tutorial-prereqs.md %}
```
Comment (jeffmendoza) [inline docs/tutorials/stateful-application/run-stateful-application.md]: Steve, The last bullet is important for this sample, I know it is covered in the first step of the tutorial, but I think it is important to call out in the prereqs.

**Comment by jeffmendoza** [comment]
@steveperry-53 Rebased and updated prereqs. Screenshot below
before you begin

**Comment by jeffmendoza** [comment]
@steveperry-53 Any other changes needed?

**Comment by erictune** [comment]
For 1.5, I would recommend that users use a StatefulSet for this use case.  StatefulSet will be beta in 1.5.

Was your plan to have this as an option in addition to StatefulSet, and if so how should users pick?  Or would this one, with the Deployment, go into the 1.4 docs, and replace it with an SS-based example for 1.5?

Or are there still users who don't want to use a beta resource?

**Comment by jeffmendoza** [comment]
The title of this one is "Running a Single-Instance Stateful Application". I was thinking we would have a complementary article "Running a Clustered Stateful Application" that covers StatefulSet. Is this correct?

**Comment by steveperry-53** [comment]
I like the idea of having this tutorial be for single instance and having another tutorial for StatefulSet.

**Comment by jeffmendoza** [comment]
Rebased

**Comment by janetkuo** [comment]
@jeffmendoza @jaredbhatti Should we cherrypick this into release-1.5 since I'm adding StatefulSets tutorial #1707 and need the navigation change added here

**Comment by jeffmendoza** [comment]
I'll leave it up to Jared.

Is there a plan for a regular merge (and conflict resolution) from master into release-1.5 to avoid a big merge at the end?

**Pull request merged**